### PR TITLE
[FR] Added asset tag to expected tags

### PR DESCRIPTION
**Pull request by eric-forte-elastic**
## Related Issue 
https://github.com/elastic/detection-rules/issues/2900

## Summary

This PR adds the `Asset: .*` tag to the list of the expected tags. The goal is for the user to be able to specify an asset e.g. `Asset: MyCloudMachineOne` and tag a rule as specific to protecting the user defined asset and still be able to use the unit tests without having to fork the repo. This PR also adds unit test support for adding other regex wildcard tags. For more detail, please see the related issue.  

## Testing

Try adding an asset tag to any given rule and running the unit tests. 

E.g. updating `rules/windows/defense_evasion_disable_windows_firewall_rules_with_netsh.toml`'s tags to be:
```
tags = [
    "Asset: AssetOne",
    "Domain: Endpoint",
    "OS: Windows",
    "Use Case: Threat Detection",
    "Tactic: Defense Evasion",
    "Resources: Investigation Guide",
    "Data Source: Elastic Endgame",
    "Data Source: Elastic Defend"
]

```

Then use the environment variable to disable tag validation via:

```shell
export DR_BYPASS_TAGS_VALIDATION=True
```

### CODE REVIEW

**Comment by eric-forte-elastic** [comment]
Need to add a check making sure that there are not empty Asset tags in the pre-built rules packages.

**Review by terrancedejesus — APPROVED** [review]
Seems fine and a minor addition.

File: tests/test_all_rules.py [hunk 1]
```diff
@@ -436,6 +436,40 @@ def test_no_duplicate_tags(self):
         if invalid:
             self.fail(f'Rules with duplicate tags:\n{invalid}')
 
+    def test_wildcard_tag_validation(self):
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:439]: Usually these test cases iterate over all rules. This test case may be better moved elsewhere.

File: tests/test_all_rules.py [hunk 2]
```diff
@@ -436,6 +436,40 @@ def test_no_duplicate_tags(self):
...
+        # Get tags with regex definitions form definitions.EXPECTED_RULE_TAGS
+        wildcard_tags = set([tag for tag in definitions.EXPECTED_RULE_TAGS if "*" in tag.split(":")[1]])
+        wildcard_map = {tag.split(":")[0]: tag for tag in wildcard_tags}
+
+        for rule in self.all_rules:
+            rule_tags = rule.contents.data.tags
+            for tag in rule_tags:
+                if tag.split(":")[0] in wildcard_map.keys():
+                    # Check if tag matches regex
+                    if not re.match(wildcard_map[tag.split(":")[0]], tag):
```
Comment (Mikaayenson) [inline tests/test_all_rules.py:467]: We should require a string after the `Asset: ` .

**Comment by eric-forte-elastic** [comment]
After review with DED, we have determined that this would be better accomplished via an environment variable to skip tag tests, as all tags should be valid for non-Elastic authored rules.

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Pull request merged**